File: c/basic/palindrome.c
#include<stdio.h>
int main(){
    int num,r,sum,temp;
    printf("enter an integer:\n");
    scanf("%d",&num);
    temp=num;
    sum=0;
    while(num>0){
        r=num%10;
        sum=sum*10+r;
        num=num/10;

    }
    if(temp==sum){
        printf("the number is a palindrome \n",temp);
    }
    else{
        printf("the number is not  a palindrome\n",temp);
    }
    return 0;
    }
  

File: Eclipse workspace.java/practicecode/src/digitsbeginner/palindrome.java
package digitsbeginner;

import java.util.Scanner;

public class palindrome {

	public static void main(String[] args) {
		Scanner sc=new Scanner(System.in);
		System.out.println("enter a number:");
		int num=sc.nextInt();
		 int originalnum=num,reverse=0;
		while(num!=0) {
			int remainder=num%10;
			reverse=reverse*10+remainder;
			num/=10;
			
		}if(originalnum==reverse) {
		System.out.println("palindrome");
		}

	}

}
